### ISSUE TITLE
LOD support

### ISSUE THREAD

**Issue by qian256** [keijiro/Pcx#14]
Awesome work!
Do you have any idea about how to enable Level of Detail control of point cloud rendering using your shaders? 
One way I can think of is to discard some vertices in geometry shader, simply based on their vertexID.
You are very knowledgeable with Unity rendering, what would be your suggestion to integrate the LOD of this shader with Unity?

**Comment by keijiro** [comment]
Before thinking about it, I'd like to define the premise.

- Why do you need LOD? I guess that you're experiencing slow down on your environment, but I'd like to make sure what the motive is.
- If you're experiencing slow down, how many points are there in the point cloud? Also I'd like to know the runtime environment (OS, CPU/GPU specs).

Defining the premise is important because it's not always effective to apply LOD to this kind of issues. If you're using low spec environment, LOD might work. If you're trying to render tons of point with an expensive PC, culling would be more important than LOD.

**Comment by qian256** [comment]
First, yes. My application is to render pointcloud on HoloLens. It is not super good computing device. With your point shader, it is able to render about 300k points. With disk shader, less than 100k points. Of course, under the condition that it renders at 60 fps (for good AR experience).
The point cloud can be always facing the user, will culling help in this case?

**Comment by keijiro** [comment]
Okay. In that case, tweaking the per-point LOD is possibly an effective approach. What happens when changing this line:

https://github.com/keijiro/Pcx/blob/master/Assets/Pcx/Runtime/Shaders/Disk.cginc#L83

as follows:

```
int slices = 2;
```

If this change improves the performance, tweaks to the per-point LOD might work. If there is no performance gain, another approach (culling/decimating) should be taken.

**Comment by qian256** [comment]
Hi @keijiro ,
I actually tried doing that. But because the point shader is much faster than disk shader. Reducing the number of slices for disk shader is still slower than point shader (at least 3 vertex generated for each input vertex in disk shader). That's why I came to the solution of "discarding" vertex of point shader. Especially when the point cloud is like an RGBD dense cloud, where the "discarding" of vertex does not cause visual artifacts.
